extensions/firestore-sheets-sync/extension.yaml

```diff
@@ -0,0 +1,119 @@
+name: sync-firestore-sheets
+version: 0.1.0
+specVersion: v1beta
+
+displayName: Sync Firestore to Google Sheets
+
+description: >-
+  Export defined collection to a Google Sheet.
+
+license: Apache-2.0
+sourceUrl: TBD
+
+author:
+  authorName: Invertase
+  url: https://invertase.io
+
+billingRequired: true
+
+apis:
+  - apiName: sheets.googleapis.com
+    reason: Powers all Sheets tasks performed by the extension.
+
+roles:
+  - role: datastore.owner
+    reason: Allows the extension to export (user) data from Cloud Firestore.
+
+resources:
+  - name: googleSheetSync
+    type: firebaseextensions.v1beta.function
+    description:
+      Listens for document changes in your specified Cloud Firestore collection,
+      then exports the changes into a Google Sheet.
+    properties:
+      location: ${param:LOCATION}
+      runtime: nodejs14
+      eventTrigger:
+        eventType: providers/cloud.firestore/eventTypes/document.write
+        resource: projects/${param:PROJECT_ID}/databases/(default)/documents/${param:COLLECTION_PATH}/{documentId}
+
+params:
+  - param: SHEET_ID
+    label: Google Sheet ID
+    description: >
+      What is the ID of the Google Sheet you want to export to?
+    type: string
+    required: true
+
+  - param: COLLECTION_PATH
+    label: Firestore collection to record export metadata
+    description: >
+      What is the path to the collection that will be synced to your Google Sheet?
+    type: string
+    default: sheets
+    example: sheets
+    required: false
+    validationRegex: '^[^/]+(/[^/]+/[^/]+)*$'
+    validationErrorMessage: Must be a valid Cloud Firestore collection
+
+  - param: FIELDS_TO_SYNC
+    label: Fields to sync
+    description: >
+      A list of comma-separated fields to sync to the Google Sheet. If left blank, no fields will be synced.
```

Comment:
Does it not make more sense to Sync all fields if left empty?

Reply:
Required since Firestore schemaless, making the user input these fields ensures only those fields are synced, since otherwise multiple records with different fields may not work since we only have a single header row in the sheet